### PR TITLE
fix(input-date-picker): range end date is set to end of the day

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #5027 

This will fix the `endDate` time in the `date-picker` and `input-date-picker` components which is now set to `23.59.59.999` local time .

### CODE REVIEW

File: src/components/date-picker/date-picker.tsx [hunk 1]
```diff
@@ -193,7 +200,7 @@ export class DatePicker {
   // --------------------------------------------------------------------------
   connectedCallback(): void {
     if (Array.isArray(this.value)) {
-      this.valueAsDate = this.value.map((v) => dateFromISO(v));
+      this.valueAsDate = mapValueToValueAsDate(this.value);
```
Comment (driskull) [inline src/components/date-picker/date-picker.tsx:203]: nitpick: I'm not sold on the name of this utility function.

I actually think it was more readable before :D

File: src/utils/date.ts [hunk 2]
```diff
@@ -212,3 +216,14 @@ export function getDaysDiff(date1: Date, date2: Date): number {
...
   return (ts1 - ts2) / (1000 * 3600 * 24);
 }
+
+/**
+ * Set time of the day to the end.
+ *
+ * @param {Date} date Date.
+ * @returns {Date} Date with time set to end of day .
+ */
+export function setEndOfDay(date: Date): Date {
```
Comment (driskull) [inline src/utils/date.ts:226]: THE END OF DAYS

**Review by driskull — APPROVED** [review]
🔥

File: src/components/date-picker/date-picker.tsx (hunk 1 above)
Comment (anveshmekala) [reply src/components/date-picker/date-picker.tsx:203]: How about `getValueAsDate` ?

Comment (driskull) [reply src/components/date-picker/date-picker.tsx:203]: Maybe `getValueAsDateRange`

**Pull request merged**